Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
+                    }
+                    if (Locomotive.ThrottlePercent < 2 && controllerBinding == ControllerBinding.Throttle)
+                    {
+                        if (Locomotive.ThrottlePercent != 0)
+                            Locomotive.SetThrottlePercent(0);
+                    }
+                    if (Locomotive.ThrottlePercent > 1 && controllerBinding == ControllerBinding.Throttle)
+                    {
+                        Locomotive.SetThrottlePercent(Locomotive.ThrottlePercent - 1f);
+                    }
+                    if (Locomotive.ThrottlePercent > 100 && controllerBinding == ControllerBinding.Throttle)
+                    {
+                        Locomotive.ThrottlePercent = 100;
+                    }
+
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncrease)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
+                    }
+                    if (Locomotive.DynamicBrakePercent == -1) Locomotive.SetDynamicBrakePercent(0);
+                    if (Locomotive.ThrottlePercent < 1 && Locomotive.DynamicBrakePercent < 100)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent + 2f);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncreaseFast)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
```

Comment:
This is the third time the same code appears, can the duplication be removed?

Reply:
changed to enum comparison